stringutil/stringutil.go

```diff
@@ -82,5 +82,12 @@ func SplitN(s, sep string, f []string) int {
 // Be aware that the byte slice and the string share the same memory - which makes
 // the string mutable.
 func ByteSlice2String(b []byte) string {
-	return *(*string)(unsafe.Pointer(&b))
+	return unsafe.String(unsafe.SliceData(b), len(b))
+}
+
+// String2ByteSlice converts a string into a byte slice without a heap allocation.
+// Be aware that the byte slice and the string share the same memory - which makes
+// the string mutable.
+func String2ByteSlice(s string) []byte {
+	return unsafe.Slice(unsafe.StringData(s), len(s))
```

Comment:
Why adding this unused function?
Just though about getting rid of ByteSlice2String() (and thus "unsafe") without adding heap allocations.

Reply:
Oh, GH folded the usage away from me... nevermind then.

Reply:
Its used exactly once at https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/502/files#diff-bbbc8290c1869327230357b1c9470fd453f5bb8b1cdfb4da652b65b65e52cd59R76.

Could probably just inline it.